prow/git/v2/client_factory.go

```diff
@@ -147,6 +147,8 @@ func NewClientFactory(opts ...ClientFactoryOpt) (ClientFactory, error) {
 			host:     o.Host,
 			username: o.Username,
 		}
+	} else if o.CookieFilePath != "" {
+		remote = &cloneURIResolverFactory{}
```

Comment:
I was expecting us to avoid making assumptions about the remote like this and to only use cloneURI rather than only using org+repo. (From the [thread here](https://github.com/kubernetes/test-infra/pull/27373#issuecomment-1244412265).)